pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -6920,6 +6940,73 @@ def infrastructure_timing(model):
             )
 
 
+def seismicity_risk(model):
+    sites = model.df_sets["SWDSites"]
+    deep = model.df_parameters["SWDDeep"]
+    pressure = model.df_parameters["SWDAveragePressure"]
+    prox = {
+        "orphan": model.df_parameters["SWDProxPAWell"],
+        "inactive": model.df_parameters["SWDProxInactiveWell"],
+        "eq": model.df_parameters["SWDProxEQ"],
+        "fault": model.df_parameters["SWDProxFault"],
+        "hplp": model.df_parameters["SWDProxHpOrLpWell"],
+    }
+    risk_factors = model.df_parameters["SWDRiskFactors"]
+
+    risk_factor_names = {
+        "orphan": "orphan_well",
+        "inactive": "inactive_well",
+        "eq": "EQ",
+        "fault": "fault",
+        "hplp": "HP_LP",
+    }
+
+    distance_risk_factors = {}
+    severity_risk_factors = {}
+    sum_risk_factors = {
+        "distance": 0,
+        "severity": 0,
+    }
+    for factor, name in risk_factor_names.items():
+        distance_risk_factor = distance_risk_factors[factor] = risk_factors[f"{name}_distance_risk_factor"]
+        severity_risk_factor = severity_risk_factors[factor] = risk_factors[f"{name}_severity_risk_factor"]
+        sum_risk_factors["distance"] += distance_risk_factor
+        sum_risk_factors["severity"] += severity_risk_factor
+
+    for factor in risk_factor_names:
+        distance_risk_factors[f"norm_{factor}"] = distance_risk_factors[factor] / sum_risk_factors["distance"]
+        severity_risk_factors[f"norm_{factor}"] = severity_risk_factors[factor] / sum_risk_factors["severity"]
+
+    pressure_thresholds = {
+        "max": risk_factors["HP_threshold"],
+        "min": risk_factors["LP_threshold"],
+    }
+
+    max_risk_factor_for_lowest_risk = 0
+    for factor in risk_factor_names:
+        norm_factor = f"norm_{factor}"
+        max_risk_factor_for_lowest_risk += distance_risk_factors[factor] * distance_risk_factors[norm_factor] * severity_risk_factors[norm_factor]
+
+    risk_metrics = {}
+    sites_included = {}
+    for site in sites:
+        weighted_site_risk_factor = 0
+        for factor in risk_factor_names:
+            if factor in ("orphan", "inactive") and deep[site]:
+                site_risk_factor = distance_risk_factors[factor]
+            else:
+                site_risk_factor = min(prox[factor][site], distance_risk_factors[factor])
+            norm_factor = f"norm_{factor}"
+            weighted_site_risk_factor += site_risk_factor * distance_risk_factors[norm_factor] * severity_risk_factors[norm_factor]
+        inverse_weighted_site_risk_factor = weighted_site_risk_factor / max_risk_factor_for_lowest_risk
+        risk_metrics[site] = 1 - inverse_weighted_site_risk_factor
+        # only consider these sites with True sites_included!
+        sites_included[site] = pressure_thresholds["min"] <= pressure[site] < pressure_thresholds["max"]
```

Comment:
```suggestion
        sites_included[site] = pressure_thresholds["min"] <= pressure[site] <= pressure_thresholds["max"]
```
Is there a reason to use different conventions for the lower and upper threshold? I think it makes sense to use the same convention for both.

Reply:
That's based on Laura's logic in her Excel.
* Is average pressure around well <.5 psi/ft?  (0=no, 1= yes) (for nearby similar - deep or shallow -  wells)
* Is average pressure around well >=.7  psi/ft?  (0=no, 1= yes) (for nearby similar - deep or shallow -  wells)

If any of the above is 1, excluded. That's .5 psi/ft (min) <= pressure < .7 psi/ft means included.

Reply:
@HuidaeCho I understand the logic, but why use `<=` when checking the lower threshold but `<` when checking the upper threshold? This is a minor thing, since it only has an effect if the pressure is exactly 0.5 or 0.7, but I think it would make more sense to be consistent when checking both thresholds. I would choose to use `<=` for both,  but using `<` for both would be fine as well.